### PR TITLE
[Service Connector] `az aks connection list/show`: Add kubernetes resource name

### PR DESCRIPTION
**Pull request by wchigit**
**Related command**
az aks connection list
az aks connection show

**Description**
This PR adds kubernetes resource name to align with Portal behavior. Customers can use `kubernetesResourceName` as key to query in a linker result. 

**Testing Guide**
az aks connection list --source-id {aks-cluster-id}
az aks connection show --id {linker-id}

**History Notes**

[Service Connector] `az aks connection list/show': Add kubernetes resource name

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py [hunk 1]
```diff
@@ -547,3 +548,25 @@ def get_secret_type_for_update(authInfo):
...
+
+
+# Decorator for AKS configurations.
+def is_aks_linker_by_id(resource_id):
+    pattern = r'/subscriptions/([^/]+)/resourceGroups/([^/]+)/providers/Microsoft.ContainerService' + \
+        r'/managedClusters/([^/]+)/providers/Microsoft.ServiceLinker/linkers/([^/]+)'
+    return re.match(pattern, resource_id, re.IGNORECASE) is not None
+
+
+def get_aks_resource_name(linker):
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py:560]: return a list

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py [hunk 2]
```diff
@@ -547,3 +548,25 @@ def get_secret_type_for_update(authInfo):
...
+    secret_name = get_aks_resource_secret_name(linker["name"])
+    if linker["authInfo"] is not None and linker["authInfo"].get("authType") == "userAssignedIdentity" and \
+            not (linker["targetService"]["resourceProperties"] is not None and
+                 linker["targetService"]["resourceProperties"].get("connectAsKubernetesCsiDriver")):
+        service_account_name = f'sc-account-{linker["authInfo"].get("clientId")}'
+        return f'{secret_name} / {service_account_name}'
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py:566]: list

**Comment by wchigit** [comment]
@kairu-ms Could you help review this PR and move this to the May milestone?

**Review by houk-ms — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**